framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -348,6 +363,13 @@ protected BaseTypeVisitor(BaseTypeChecker checker, @Nullable Factory typeFactory
                 checker.hasOption("assumeDeterministic") || checker.hasOption("assumePure");
         assumePureGetters = checker.hasOption("assumePureGetters");
         checkCastElementType = checker.hasOption("checkCastElementType");
+        lintCastEnabled = checker.getLintOption("cast", false);
+        lintCastRedundantEnabled =
+                checker.getLintOption("cast:redundant", false) || lintCastEnabled;
```

Comment:
```suggestion
        lintCastRedundantEnabled =
                lintCastEnabled || checker.getLintOption("cast:redundant", false);
```

This avoids looking at the lint option if the answer is already clear. Do the same for the others.

I guess what doesn't work is `-Alint=cast,-cast:redundant`... To enable the overall category, but then disable some options. Can you think how this could be done?

Reply:
The new version works because the [getLintOption](https://github.com/eisop/checker-framework/blob/8f2e3f95ef3b55a29763a56b0ba50e3cc51d6945/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java#L1967-L1993) logic will look up its parent option, for example `cast` is the parent option for `cast:redundant`. When the child option is not supplied, the code will look for whether the parent option is supplied. Otherwise, it will return the default defined by the second argument.

As a result, we don't need to store the `cast` and `instanceof` options as separate fields.